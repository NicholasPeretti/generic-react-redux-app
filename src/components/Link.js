import React from 'react'
import PropTypes from 'prop-types'
import LinkStyle from '../utils/styledComponents/link-style'
import styled from 'styled-components'
import { Link as RouterLink } from 'react-router-dom'

const StyledLink = styled(RouterLink)`
  ${LinkStyle};
`

var Link = ({ to, children }) => (
  <StyledLink to={to}>
    {children}
  </StyledLink>
)

Link.propTypes = {
  to: PropTypes.string.isRequired
}

export default Link
