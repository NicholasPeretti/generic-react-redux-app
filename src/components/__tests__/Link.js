import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import Link from '../Link'
import { MemoryRouter } from 'react-router-dom'

const WrappedLink = (props) => (
  <MemoryRouter>
    <Link {...props} />
  </MemoryRouter>
)

describe('Link', () => {
  PresentationalTest(WrappedLink, { to: '/' })

  test('Should match snapshot', () => {
    const wrapper = shallow(<WrappedLink to='/page1'>This is a Link</WrappedLink>)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
