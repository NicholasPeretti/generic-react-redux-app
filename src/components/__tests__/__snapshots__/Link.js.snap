// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should match snapshot 1`] = `"<a class=\\"sc-bdVaJa enetwA\\" href=\\"/page1\\">This is a Link</a>"`;
